sequencer/src/block2.rs

```diff
@@ -0,0 +1,144 @@
+// Question: jellyfish uses `ark_std` everywhere instead of `std`.
+// Is there a chance of badness if we (as a downstream user) use `std`?
+use std::mem::size_of;
+
+#[allow(dead_code)] // TODO temporary
+pub struct BlockPayload {
+    payload: Vec<u8>,
+    // tx_ranges: Vec<Range<u32>>, // not convinced we need this
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+pub struct Tx {
+    // namespace: u32, // TODO how many bytes for namespace id?
```

Comment:
We should be able to reuse the `Transaction` struct from `transaction.rs` here. It's basically exactly this. But we have a `VmId` newtype for the namespace.

Reply:
I knew that. I was just testing.

We may wish to change it from `u64` to `u32` though. I don't anticipate ever having more than 4.2 billion namespaces.

Reply:
Yeah I've been thinking about that. While there will never be more than 2^32 distinct namespaces, it is nice to be able to use a larger integer for the namespace ID. For example, for Vienna we use 0xc0ffee1 just for fun, which happens to fit in 4 bytes but you could imagine a similar idea that doesn't. Someone in Discord also suggested using the output of a hash function as a namespace ID, for example the hash of the rollup name (and indeed, Ethereum chain IDs are 32 bytes).

However, we can always just have a separate mapping from long namespace IDs to short ones, if that's desirable, so I think going with 4 bytes for now is fine.